core/src/consensus/tendermint/worker.rs

```diff
@@ -971,8 +972,9 @@ impl Worker {
 
         let current_height = self.height;
         let vote_step = VoteStep::new(self.height, self.view, self.step.to_step());
-        let proposal_at_current_view = self.votes.get_block_hashes(&vote_step).first().cloned();
-        if proposal_at_current_view == Some(proposal.hash()) {
+        let is_current_proposal =
+            self.votes.get_block_hashes(&vote_step).into_iter().any(|block_hash| block_hash == proposal.hash());
+        if is_current_proposal {
```

Comment:
Putting a verb prefix in a local boolean variable is awkward.
Microsoft's conventions "suggests" that boolean "object property" can have 'is', 'has', 'can' prefix[1]. However, the convention is not blindly applicable to another context.
Local boolean variables don't have an "object" to provide contexts. Also when the method/field is not only about an object that contains it, verb prefix becomes awkward [2]
So, `current_proposal_exists` would be more readable name.

[1] https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-properties
[2] https://softwareengineering.stackexchange.com/a/102738

Reply:
I want to check that the `proposal` variable is for the current step. `current_proposal_exists` seems there is another proposal in the current step. How about `current_proposal` as the variable name.